### PR TITLE
Release 1.14.2: update release notes, upgrade notes and latest version

### PR DESCRIPTION
**Pull request by inteon**
Updates:
- references to latest cert-manager version
- release notes
- upgrade notes

### CODE REVIEW

File: content/docs/releases/release-notes/release-notes-1.14.md [hunk 1]
```diff
@@ -14,12 +14,25 @@ support for creating [CA certificates with "Name Constraints" and "Authority Inf
...
+## `v1.14.2`
+
+> 📢 When upgrading to cert-manager release 1.14, please skip `v1.14.0` and `v1.14.1` and install this patch version instead.
+
+### Changes since `v1.14.1`
+
+#### Bug or Regression
+
+- BUGFIX: cert-manager CA and SelfSigned issuers incorrectly copied the critical flag from the CSR instead of re-calculating that field themselves. (#6727, @jetstack-bot)
+- Helm: Fix a bug in the logic that differentiates between 0 and an empty value. (#6729, @jetstack-bot)
```
Comment (wallrj) [inline content/docs/releases/release-notes/release-notes-1.14.md:26]: The PR and authors should be markdown links.
> Replace the GitHub issue numbers and GitHub handles (e.g., #1234 or @maelvls) with actual links using the following command:
https://cert-manager.io/docs/contributing/release-process/#minor-releases

**Review by wallrj — CHANGES_REQUESTED** [review]

File: content/docs/releases/release-notes/release-notes-1.14.md [hunk 2]
```diff
@@ -14,12 +14,25 @@ support for creating [CA certificates with "Name Constraints" and "Authority Inf
...
+## `v1.14.2`
+
+> 📢 When upgrading to cert-manager release 1.14, please skip `v1.14.0` and `v1.14.1` and install this patch version instead.
+
+### Changes since `v1.14.1`
+
+#### Bug or Regression
+
+- BUGFIX: cert-manager CA and SelfSigned issuers incorrectly copied the critical flag from the CSR instead of re-calculating that field themselves. (https://github.com/cert-manager/cert-manager/pull/6727, https://github.com/jetstack-bot)
+- Helm: Fix a bug in the logic that differentiates between 0 and an empty value. (https://github.com/cert-manager/cert-manager/pull/6729, https://github.com/jetstack-bot)
```
Comment (wallrj) [inline content/docs/releases/release-notes/release-notes-1.14.md:26]: Need to be Markdown links. 
image

**Review by wallrj — CHANGES_REQUESTED** [review]

File: content/docs/releases/release-notes/release-notes-1.14.md (hunk 2 above)
Comment (inteon) [reply content/docs/releases/release-notes/release-notes-1.14.md]: Sorry for doing this wrong a second time.

**Review by wallrj — APPROVED** [review]
/lgtm

**Pull request merged**